internal/cephfs/nodeserver.go

```diff
@@ -254,16 +254,11 @@ func (ns *NodeServer) NodeUnpublishVolume(ctx context.Context, req *csi.NodeUnpu
 	if err = util.ValidateNodeUnpublishVolumeRequest(req); err != nil {
 		return nil, err
 	}
-
-	volID := req.GetVolumeId()
+	// notice: we have been operating here with a lock while unmounting the volume
+	// which slowdown the unmount operation when there are more pods and delete performed
+	// at almost same time. Considering we have enough safe-guards on CO side, we can
+	// operate outside the critical section.
```

Comment:
This comment is not needed, and  probably more confusing than it can be. No need to mention issues (slowdown), just explain *why* it is safe to not use locking here.

It would be better to explain the working of NodeUnpublishVolume in the comment above the function, and maybe mention that locking is not required (why?) for unmounting.

Reply:
sure... let me address this in a detail or change the comment with the sequence ..etc